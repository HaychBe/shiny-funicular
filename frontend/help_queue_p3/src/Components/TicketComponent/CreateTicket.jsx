import React, {useState } from "react";
import axios from "axios";
import { Button, Modal, Form} from "react-bootstrap";
import CreateTicketImg from "../../css/images/CreateTicket.png"

const CreateTicket = ({refreshPage}) => {
    
    const [show, setShow] = useState(false);

    const handleClose=()=> setShow(false);
    const handleShow=()=> setShow(true);


    const [traineeName, setTraineeName] = useState(``);
    const [cohort, setCohort] = useState(``);
    const [topic, setTopic] = useState(``);
    const [trainer, setTrainer] = useState(``);
    const [title, setTitle] = useState(``);
    const [description, setDescription] = useState(``);
    const [time, setTime] = useState(
        new Date().toLocaleString('en-GB')
    );

    const ticketValues = {
        title: title,
        topic: topic,
        trainee: traineeName,
        cohort: cohort,
        trainer: trainer,
        _description: description,
        dateTime: time,
    };

    const sendData = (i) => {
        const newDate = new Date().toLocaleString('en-GB')
    setTime(newDate);

    axios
        .post(`http://localhost:8080/ticket/create`, ticketValues)
        .then(function (response) {
           console.log(response);
           refreshPage(true);
        })
        .catch(function (error) {
           console.log(error);
        })
        handleClose();
    };

    return(
        <>
            <Button
                id="createBtn" 
                variant='light'
                onClick={handleShow}><img src={CreateTicketImg} alt="createTicketBtn"/>    
            </Button>

        <Modal
            show={show}
            onHide={handleClose}
            dialogClassName="createModal"
            scrollable={true}>
                <Modal.Header closeButton>
                    <Modal.Title>Create Ticket Modal</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    <Form>
                        <Form.Group>
                            <Form.Label>Trainee Name</Form.Label>
                            <Form.Control
                                type="text"
                                placeholder="Trainee Name"
                                onChange={(i) => {
                                    setTraineeName(i.target.value);
                                }}
                            />
                        </Form.Group>

                        <Form.Group>
                            <Form.Label>Cohort</Form.Label>
                            <Form.Control
                                as="select"
                                onChange={(i) => {
                                    setCohort(i.target.value);
                                    }}
                            >
                                <option value ="" selected disabled hidden>Choose here</option>
                                <option>CloudNative</option>                        
                                <option>Software Specialist</option>                        
                                <option>DevOps</option>   
                            </Form.Control>                     
                        </Form.Group>

                        <Form.Group>
                            <Form.Label>Topics</Form.Label>
                            <Form.Control
                                as="select"
                                onChange={(i) => {
                                    setTopic(i.target.value);
                                }}
                            >
                                <option value ="" selected disabled hidden>Choose here</option>
                                <option>FrontEnd</option>
                                <option>BackEnd</option>
                                <option>DevOps</option>
                                <option>Cloud Fundamentals</option>
                            </Form.Control>
                        </Form.Group>

                        <Form.Group>
                        <Form.Label>Trainer</Form.Label>
                            <Form.Control
                                as="select"
                                onChange={(i) => {
                                    setTrainer(i.target.value);
                                }}
                            >
                                <option value ="" selected disabled hidden>Choose here</option>
                                <option>Vinesh Ghela</option>
                                <option>Jordan Harrison</option>
                                <option>Reece Elder</option>
                                <option>Nathan Forester</option>
                                <option>Harry Volker</option>
                            </Form.Control>
                        </Form.Group>

                        <Form.Group>
                            <Form.Label>Title</Form.Label>
                            <Form.Control
                                type="text"
                                placeholder="Topic issue"
                                onChange={(i)=> {
                                    setTitle(i.target.value);
                                }}
                            />
                        </Form.Group>
                        
                        <Form.Group>
                            <Form.Label>Description</Form.Label>
                            <Form.Control
                                as="textarea"
                                rows={5}
                                onChange={(i) => {
                                    setDescription(i.target.value);
                                }}
                            />
                        </Form.Group>
                    </Form>

                    <Button
                        onClick={(i) => sendData(i)}
                        variant="success"
                        size="lg"
                        block
                    > Submit Ticket

                    </Button>
                </Modal.Body>
            </Modal>
        </>
    )
}

export default CreateTicket ;


