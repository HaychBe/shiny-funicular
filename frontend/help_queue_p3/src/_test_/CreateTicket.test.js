import React from 'react';
import renderer from 'react-test-renderer';
import CreateTicket from  '../Components/TicketComponent/CreateTicket';





it('renders correctly', () => {
    const title = "Jest ";
    const topic = "Jest Test ";
    const traineeName = "Shamsi";
    const cohort = "cne";
    const trainer = "Reece";
    const description = "Jest Test description";
    const time = "0505";
    const tree = renderer.create(<CreateTicket title={title} topic={topic} traineeName={traineeName} cohort={cohort} trainer={trainer} description={description} time={time}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

