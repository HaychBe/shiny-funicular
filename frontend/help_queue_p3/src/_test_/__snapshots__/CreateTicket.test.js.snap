// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create should match the snapshot every time it is rendered 1`] = `
<button
  className="btn btn-success"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Create Ticket
</button>
`;

exports[`CreateTicket should match the snapshot every time it is rendered 1`] = `
<button
  className="btn btn-success"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Create Ticket
</button>
`;

exports[`renders correctly 1`] = `
<button
  className="btn btn-success"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Create Ticket
</button>
`;
